### DISCUSSION TITLE
How to load kernel modules by default?

### DISCUSSION THREAD

**Discussion by DjVinnii** [kube-hetzner/terraform-hcloud-kube-hetzner#1211] [Q&A]
I deployed Rook instead of Longhorn as a storage operator in my cluster. I noticed some pods are in a CrashLoopBackOff state. After some investigation I found out this is caused by some pods that are trying to enable the `rbd` kernel module. Apparently the root cause is the mismatch of the modprobe version on the host versus the container. The newer version on the host ships zstd-compressed kernel modules, but the container itself does not support that. 

As a workaround I would like to enable the kernel module by default (on boot) on the agents until the compressed formatted is supported in the upstream container images. Is there a recommended way to achieve this?

Host:
```
# grep PRETTY /etc/os-release
PRETTY_NAME="openSUSE MicroOS"
# modprobe --version
kmod version 31
+ZSTD +XZ +ZLIB +LIBCRYPTO -EXPERIMENTAL
```

Container:
```
# grep PRETTY /etc/os-release
PRETTY_NAME="CentOS Stream 8"
# modprobe --version
kmod version 25
+XZ +ZLIB +OPENSSL -EXPERIMENTAL
```

**Comment by pat-s** [comment]
The following worked for me:

```
echo -e "rbd\nnbd" > /etc/modules-load.d/custom.conf
```

Reboot afterward, and modules are loaded by default on the host. This lets the rbd and nbd pods succeed.
